4 warcaby jakub

MrJKL opened the pull request.

PiotrLarkowski committed:
Merge pull request #16 from aPusio/3-umieszczenie-menu-w-pętli

3-umieszczenie-menu-w-pętli - Umieszczenie Menu w pętli

MrJKL committed:
4-warcaby - Utworzenie klasy Checkers i entity Player

MrJKL committed:
4-warcaby - Utworzenie Entity DAO

MrJKL committed:
4-warcaby - Utworzenie PlayerDao PlayerEntity i pomocnicze

MrJKL committed:
4-warcaby - Utworzenie Encji i zmiana HibernateFactory

JanczarekPawel committed:
dodanie encji

JanczarekPawel committed:
Merge pull request #31 from aPusio/FeatureGame

dodanie encji

JanczarekPawel committed:
dodanie encji - poprawki

JanczarekPawel committed:
dodanie encji - poprawki

MrJKL committed:
4-warcaby - baza bez dependencji

MrJKL committed:
4-warcaby - dodanie relacji do bazy

MrJKL committed:
4-warcaby - merge od mastera

MrJKL committed:
4-warcaby - usunięcie klasy Player - konflikt nazw

MrJKL committed:
4-warcaby - drobny reformat

MrJKL committed:
4-warcaby - dodanie kilku klas

MrJKL committed:
4-warcaby - dodanie kilku dao

MrJKL committed:
4-warcaby - dodanie kilku dao

MrJKL committed:
4-warcaby - dodanie kilku dao

MrJKL committed:
4-warcaby - dodanie kilku dao

MrJKL committed:
4-warcaby - test zapisu do bazy

MrJKL committed:
4-warcaby - wyświetlanie

MrJKL committed:
4-warcaby - wyświetlanie

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
4-warcaby - prace nad zapisywaniem

MrJKL committed:
Merge remote-tracking branch 'origin/master' into 4-warcaby-jakub

MrJKL committed:
4-warcaby - ukończenie zapisu do bazy

MrJKL committed:
Merge branch 'master' into 4-warcaby-jakub

MrJKL committed:
4-warcaby - rozwijanie funkcji zapisu

MrJKL committed:
Merge remote-tracking branch 'origin/4-warcaby-jakub' into 4-warcaby-jakub

# Conflicts:
#	src/main/java/com/sda/zdjavapol107/App.java

MrJKL committed:
4-warcaby - rozwijanie funkcji zapisu

MrJKL committed:
4-warcaby - reformat

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/Pawn.java:
```diff
@@ -0,0 +1,51 @@
+
+@AllArgsConstructor
+@Getter
+public class Pawn {
```
mozesz pomyslec w tym przypadku o dziedziczenu. 
Zrobic interfejst/klase abstrakcyjna Pawn i do tego 2 implementacje normalna i damka :) bo takich ifow zacznie tylko dochodzic

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java:
```diff
@@ -0,0 +1,82 @@
+        Map<Integer, GamePeaces> gamePeacesMap = createGamePeacesMap(checkersBoard, game, playerOne, playerTwo);
+        Set<Move> setOfMoves = createSetOfMoves(checkersBoard, gamePeacesMap, playerOne, playerTwo, game);
+
+        SessionFactory sessionFactory = new HibernateFactory().getSessionFactory();
```
przekazuj sessionFactory poprzez konstruktor, nie tworz nowego :D

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java:
```diff
@@ -0,0 +1,82 @@
+        return game;
+    }
+
+    private Map<Integer, GamePeaces> createGamePeacesMap(CheckersBoard checkersBoard, Game game, Player playerOne, Player playerTwo) {
```
z tymi mapami wydaje mi sie jakies przekombinowane, ale na razie mozg mi paruje :P musze ogarnac jakas lepsza alternatywe albo przynajmniej pomysl jak to ladniej podizelic

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,56 @@
+        Long id = 0L;
+        Game game = null;
+        for (Game g : gameList) {
+            if (g.getId() > id) {
```
nie opieral bym sie o id w wyznaczaniu ostatniej gry, bezpieczniej bylo by zapisywac jakis timestamp kiedy gra sie odbyla :)

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,56 @@
+    private List<CheckersPlayer> convertPlayersToChPlayers (List<Player> players) {
+        List<CheckersPlayer> checkersPlayerList = new ArrayList<>();
+        char playerOneColor = players.get(0).getGamePeaces().stream()
+                .findAny().get().getColor();
```
findAny zwraca optional bo teoretycznie mozesz nie znalezc wyniku, jestes pewien ze to nigdy nie rzuci wyjatkiem ?

aPusio replied:
ogolnie mozesz to napisac w zupelnie inny sposob, jesli kolor jest present dodaj go do listy :)

MrJKL replied:
dla uproszczenia dodałem do encji player pole z kolorem

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersBoard.java:
```diff
@@ -0,0 +1,64 @@
+            for (int j = 0; j < 8; j++) {
+                if (i != 3 && i != 4) {
+                    if (i % 2 == 0 && j % 2 == 0) {
+                        result[i][j] = pawnMap.get(keyValue);
```
czy te 3 linijki tutaj i nizej sa takie same ? 
jesli tak moze zamknac je w jakijs funkcji :) ?

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersBoard.java:
```diff
@@ -0,0 +1,64 @@
+
+    private Pawn[][] generateNewBoard() {
+        pawnMap = new HashMap<>();
+        for (int i = 0; i < 24; i++) {
```
stale jak 24 albo nizej 8 staraj sie trzymac w zmiennych statycznych finalnych i wtedy je uzywac. gdyby jakims cudem ktorys z tych rozmiarow sie zmienil to zmieniasz to tylko w jednym miejsciu :) no i opisanie tego slownie np BOARD_SIZE, PAWN_AMOUNT czy cos w tym stylu jest czytelniejsze niz sama liczba ktora nic nie mowi

aPusio commented:
moze tej logiki nie ma tutaj za wiele ale encje i customowe metody w dao wyszly pieknie !

MrJKL committed:
Poprawki:
- dodanie stałych w miejsce nieopisanych wartości liczbowych
- przekazywanie sessionfactory z app.java
- wybieranie ostatniego zapisu na podstawie daty i czasu

MrJKL committed:
Poprawki:
- dodanie pola koloru do encji player i związane z tym poprawki kodu

aPusio commented on src/main/java/com/sda/zdjavapol107/HibernateFactory.java:
```diff
@@ -17,6 +18,13 @@ private org.hibernate.cfg.Configuration getHibernateConfig() {
+        configuration.addAnnotatedClass(Move.class);
+        configuration.addAnnotatedClass(Player.class);
+        configuration.addAnnotatedClass(Tournament.class);
+
```
nadmiarowy enter :)

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersBoard.java:
```diff
@@ -0,0 +1,66 @@
+    private CheckersPlayer playerTwo;
+    private CheckersTournament checkersTournament;
+    private CheckersTurn turn;
+
```
nadmiarowy enter :)

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,62 @@
+
+public class CheckersLoad {
+
+    SessionFactory sessionFactory;
```
private

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,62 @@
+        CheckersTournament checkersTournament = new CheckersTournament(tournament.getPlayerOneScore(), tournament.getPlayerTwoScore());
+//        List<CheckersPlayer> checkersPlayerList = convertPlayersToChPlayers(playerList);
+
+        return null;
```
rozumiem ze to jest TODO :D ?

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,62 @@
+        return null;
+    }
+
+    private Game loadLatestGame(List<Game> gameList) {
```
to jest fajne miejsce na uzycie Optionala :) 
poza tym fajnie mozna to zrobic streamem, mozesz sprobowac (chociaz to raczej w ramach treningu uzycie streama nic nie zmieni :) )

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersLoad.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    private List<CheckersPlayer> convertPlayersToChPlayers (List<Player> players) {
+        List<CheckersPlayer> checkersPlayerList = new ArrayList<>();
```
da sie troche krocej :) cos w stylu 
```java
List.of((
   new CheckersPlayer(players.get(0).getName(), players.get(0).getColor()), 
   new CheckersPlayer(players.get(1).getName(), players.get(1).getColor())
 )
```

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java:
```diff
@@ -0,0 +1,88 @@
+
+public class CheckersSave {
+
+    SessionFactory sessionFactory;
```
private

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java:
```diff
@@ -0,0 +1,88 @@
+    SessionFactory sessionFactory;
+    private final int NUMBER_OF_PAWNS = 24;
+
+
```
nadmiarowy enter :D

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java:
```diff
@@ -0,0 +1,88 @@
+        Map<Integer, GamePeaces> gamePeacesMap = createGamePeacesMap(checkersBoard, game, playerOne, playerTwo);
+        Set<Move> setOfMoves = createSetOfMoves(checkersBoard, gamePeacesMap, playerOne, playerTwo, game);
+
+        PlayerDao playerDao = new PlayerDao(sessionFactory);
```
mozesz ogolnie wszystkie dao wyrzucic do konstruktora :) 
jak chcesz wiedziec jak to zrobic dobrze w calym projekcie to poczytaj o dependency injection :)

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/PawnMove.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.Getter;
+
+@Getter
+public class PawnMove {
```
jest anotacja lombokowa ktora generuje konstruktor tylko dla parametrow final - @RequiredArgsConstructor czy jakos tak :)

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/dao/GamePeacesDao.java:
```diff
@@ -0,0 +1,25 @@
+        return gamePeacesList;
+    }
+
+
```
nadmiarowy enter :D

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/model/Move.java:
```diff
@@ -0,0 +1,37 @@
+@NoArgsConstructor
+@Getter
+@Entity
+
```
nadmiarowy enter :P

aPusio commented on src/main/java/com/sda/zdjavapol107/checkers/model/Tournament.java:
```diff
@@ -0,0 +1,35 @@
+    }
+}
+
+
```
nadmiarowe entery ^^

MrJKL committed:
Poprawki:
- drobne zmiany
